### PR TITLE
Upgrade urbanairship and xcode 10

### PR DESCRIPTION
**Pull request by ericgibb**

### CODE REVIEW

File: ios/Hoard/Info.plist [hunk 1]
```diff
@@ -9,33 +9,17 @@
...
-	<string>com.hoardinc.$(PRODUCT_NAME:rfc1034identifier)</string>
+	<string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
 	<key>CFBundleInfoDictionaryVersion</key>
 	<string>6.0</string>
 	<key>CFBundleName</key>
 	<string>$(PRODUCT_NAME)</string>
 	<key>CFBundlePackageType</key>
 	<string>APPL</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.8</string>
```
Comment (zachgibb) [inline ios/Hoard/Info.plist]: we'll  need to update this version number

File: ios/Hoard/Info.plist [hunk 2]
```diff
@@ -65,35 +51,10 @@
 			</dict>
 		</dict>
 	</dict>
-	<key>NSCameraUsageDescription</key>
-	<string>Allow access to camera to enable QR scanning support.</string>
-	<key>NSContactsUsageDescription</key>
-	<string>This app requires contacts access to send and recieve payments.</string>
```
Comment (zachgibb) [inline ios/Hoard/Info.plist]: we need these permissions

File: ios/Hoard/Info.plist [hunk 3]
```diff
@@ -56,6 +40,8 @@
 	<true/>
 	<key>NSAppTransportSecurity</key>
 	<dict>
+		<key>NSAllowsArbitraryLoads</key>
+		<true/>
```
Comment (zachgibb) [inline ios/Hoard/Info.plist]: is this necessary? we should remove this if we can. we wouldn't be able to submit the app for final review with this permission on

File: ios/Hoard/Info.plist [hunk 4]
```diff
@@ -126,10 +109,10 @@
 	<key>UISupportedInterfaceOrientations</key>
 	<array>
 		<string>UIInterfaceOrientationPortrait</string>
+		<string>UIInterfaceOrientationLandscapeLeft</string>
+		<string>UIInterfaceOrientationLandscapeRight</string>
```
Comment (zachgibb) [inline ios/Hoard/Info.plist]: i think we had decided to disallow non-portrait orientations

File: patches/metro-minify-uglify+0.30.2.patch [hunk 5]
```diff
@@ -1,12 +0,0 @@
-patch-package
---- a/node_modules/metro-minify-uglify/src/minifier.js
-+++ b/node_modules/metro-minify-uglify/src/minifier.js
-@@ -34,7 +34,7 @@ filename)
- 
- function minify(inputCode, inputMap) {
-   const result = uglify.minify(inputCode, {
--    mangle: { toplevel: true },
-+    mangle: false,
```
Comment (zachgibb) [inline patches/metro-minify-uglify+0.30.2.patch]: do we no longer need this patch?

File: src/App.js [hunk 6]
```diff
@@ -136,35 +137,145 @@ const ModalStack = createStackNavigator(
...
 );
 
-export default class App extends React.Component {
+/*
+  Example usage:
+  handleDeepLink('com.hoardinc.Hoard://confirm_transaction/?params={"tx":"123","uid":"12"}')
+ */
+
+function handleDeepLink(deepLink) {
+  const SCHEME = 'com.hoardinc.Hoard://';
```
Comment (zachgibb) [inline src/App.js]: will this need to be updated to `com.hoard`?

File: src/screens/Settings/Settings/Settings.js [hunk 7]
```diff
@@ -1,30 +1,54 @@
...
+                UrbanAirship.setUserNotificationsEnabled(value);
+                props.updateEnablePushNotifications(value);
+              }}
+              value={props.enablePushNotifications}
+            />
+          }
         />
       </Body>
+      <Footer>
+        <T.Light style={styles.debug} />
```
Comment (zachgibb) [inline src/screens/Settings/Settings/Settings.js]: what's up with this?

File: src/screens/Wallet/Confirm.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  return '';
+}
+
+export async function makeTransactionRequest(url, queries) {
+  return api.get(`${url}${makeQueryString(queries)}`);
+}
+
+export default class ConfirmTransaction extends Component {
```
Comment (zachgibb) [inline src/screens/Wallet/Confirm.js]: is this class mostly a placeholder at this point?

File: patches/metro-minify-uglify+0.30.2.patch (hunk 5 above)
Comment (ericgibb) [reply patches/metro-minify-uglify+0.30.2.patch]: correct! it's now part of the package.json configuration

File: src/screens/Wallet/Confirm.js (hunk 8 above)
Comment (zachgibb) [reply src/screens/Wallet/Confirm.js]: (yes)

**Pull request merged**